src/main/java/hudson/os/windows/ManagedWindowsServiceLauncher.java

```diff
@@ -473,13 +482,17 @@ public void afterDisconnect(SlaveComputer computer, TaskListener listener) {
             JISession session = JISession.createSession(auth);
             session.setGlobalSocketTimeout(60000);
             SWbemServices services = WMI.connect(session, determineHost(computer));
-            String id = generateServiceId(computer.getNode().getRemoteFS());
-            Win32Service slaveService = services.getService(id);
-            if(slaveService!=null) {
-                listener.getLogger().println(Messages.ManagedWindowsServiceLauncher_StoppingService(getTimestamp()));
-                slaveService.StopService();
-                listener.getLogger().println(Messages.ManagedWindowsServiceLauncher_UnregisteringService(getTimestamp()));
-                slaveService.Delete();
+            
+            Slave node = computer.getNode();
+            if (node != null) {
```

Comment:
:ant: throw `AbortException` otherwise, assuming this is something used during launch.